src/v/pandaproxy/schema_registry/storage.h

```diff
@@ -116,6 +116,15 @@ class topic_key_type_handler
 
 struct schema_key {
     static constexpr topic_key_type keytype{topic_key_type::schema};
+
+    // The record is only valid if its offset in the topic matches `seq`
+    model::offset seq;
```

Comment:
I think we could maintain read compatibility be making these new fields optional.

Reply:
Punting the "import from existing topics" functionality to https://github.com/vectorizedio/redpanda/issues/1902